### ISSUE TITLE
InflightMessageList functionality is broken in maxwell 1.40.3

### ISSUE THREAD

**Issue by lukestephenson-zendesk** [zendesk/maxwell#2041]
We just upgraded from maxwell `1.40.2` to `1.40.3` and are seeing a huge number of errors in the logs. Examples of those errors:
```
{"@timestamp":"2023-09-07T01:34:14.945955002Z","@version":"1","message":"Error executing user-provided callback on message for topic-partition 'maxwell-0'","logger_name":"org.apache.kafka.clients.producer.internals.ProducerBatch","thread_name":"kafka-producer-network-thread | maxwell","level":"ERROR","level_value":40000,"stack_trace":"java.lang.NullPointerException: Cannot assign field \"isComplete\" because \"m\" is null
   at com.zendesk.maxwell.producer.InflightMessageList.completeMessage(InflightMessageList.java:127)
   at com.zendesk.maxwell.producer.AbstractAsyncProducer$CallbackCompleter.markCompleted(AbstractAsyncProducer.java:33)
   at com.zendesk.maxwell.producer.KafkaCallback.onCompletion(MaxwellKafkaProducer.java:91)
   at org.apache.kafka.clients.producer.KafkaProducer$AppendCallbacks.onCompletion(KafkaProducer.java:1505)
   at org.apache.kafka.clients.producer.internals.ProducerBatch.completeFutureAndFireCallbacks(ProducerBatch.java:270)
   at org.apache.kafka.clients.producer.internals.ProducerBatch.done(ProducerBatch.java:234)
   at org.apache.kafka.clients.producer.internals.ProducerBatch.complete(ProducerBatch.java:180)
   at org.apache.kafka.clients.producer.internals.Sender.completeBatch(Sender.java:692)
   at org.apache.kafka.clients.producer.internals.Sender.completeBatch(Sender.java:663)
   at org.apache.kafka.clients.producer.internals.Sender.lambda$null$1(Sender.java:589)
   at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
   at org.apache.kafka.clients.producer.internals.Sender.lambda$handleProduceResponse$2(Sender.java:576)
   at java.base/java.lang.Iterable.forEach(Iterable.java:75)
   at org.apache.kafka.clients.producer.internals.Sender.handleProduceResponse(Sender.java:576)
   at org.apache.kafka.clients.producer.internals.Sender.lambda$sendProduceRequest$5(Sender.java:850)
   at org.apache.kafka.clients.ClientResponse.onComplete(ClientResponse.java:154)
   at org.apache.kafka.clients.NetworkClient.completeResponses(NetworkClient.java:594)
   at org.apache.kafka.clients.NetworkClient.poll(NetworkClient.java:586)
   at org.apache.kafka.clients.producer.internals.Sender.runOnce(Sender.java:328)
   at org.apache.kafka.clients.producer.internals.Sender.run(Sender.java:243)
   at java.base/java.lang.Thread.run(Thread.java:833)
"}
```

The most likely cause of this is #2019 which affects the use of `InflightMessageList` where this Exception is thrown.

This appears to be logged for each message published to Kafka.

**Comment by osheroff** [comment]
yeah that commit is fairly short-sighted, `InflightMessageList` works on a binlog position and the net affect of this is a bunch of rows in the message queue all with the same position.  I'm reverting and will put out a new release.

**Comment by osheroff** [comment]
fixed in 1.40.5
